monai/metrics/metric.py

```diff
@@ -140,64 +187,106 @@ def reset(self):
         self._synced_tensors = None
         self._synced = False
 
-    def add(self, *data: torch.Tensor):
+    def extend(self, *data) -> None:
         """
-        Add samples to the cumulative buffers.
+        Extend the local buffers with new ("batch-first") data.
+        A buffer will be allocated for each `data` item.
+        Compared with `self.append`, this method adds a "batch" of data to the local buffers.
 
         Args:
-            data: list of input tensor, make sure the input data order is always the same in a round.
-                every item of data will be added to the corresponding buffer.
+            data: each item can be a "batch-first" tensor or a list of "channel-first" tensors.
+                they will be concatenated at the 0-th dimension when `get_buffer()` is called.
+        """
+        if self._buffers is None:
+            self._buffers = [[] for _ in data]
+        for b, d in zip(self._buffers, data):
+            # converting to pytorch tensors so that we can use the distributed API
+            d_t: torch.Tensor
+            d_t, *_ = convert_data_type(d, output_type=torch.Tensor, wrap_sequence=True)  # type: ignore
+            try:
+                b.extend([x[0] for x in torch.split(d_t, 1, dim=0)])
```

Comment:
I think maybe `torch.unbind()` is more suitable than `torch.split` here?

Thanks.

Reply:
Not sure why we make the change, could you please clarify the advantage?

Reply:
I think `torch.split()` is mainly used for specified length of data chunks, here we fallback to use `length=1`, seems `torch.unbind` is simpler for this use case?

Thanks.

Reply:
It will change the outcome for one of the test case:
`[x[0] for x in torch.split(torch.tensor([]), 1, dim=0)]` vs `torch.unbind(torch.tensor([]), 0)`, I think the split is better here.

Reply:
OK.

Thanks.